frontend/src/metabase/query_builder/containers/QueryBuilder.unit.spec.tsx

```diff
@@ -286,7 +347,52 @@ describe("QueryBuilder", () => {
     it("should not trigger beforeunload event when query is unedited", async () => {
       const { mockEventListener } = await setup({
         card: TEST_NATIVE_CARD,
-        initialRoute: `/question/${TEST_NATIVE_CARD.id}`,
+      });
+
+      const mockEvent = callMockEvent(mockEventListener, "beforeunload");
+      expect(mockEvent.preventDefault).not.toHaveBeenCalled();
+      expect(mockEvent.returnValue).not.toEqual(BEFORE_UNLOAD_UNSAVED_MESSAGE);
+    });
+  });
+
+  describe("beforeunload events in structured queries", () => {
```

Comment:
This new set of tests along with https://github.com/metabase/metabase/blob/4217c360cfe5770c1cc28e766bf6d926a4aabbe6/frontend/src/metabase/query_builder/containers/QueryBuilder.unit.spec.tsx#L301

will ensure the new selector is tested thoroughly
https://github.com/metabase/metabase/blob/4217c360cfe5770c1cc28e766bf6d926a4aabbe6/frontend/src/metabase/query_builder/selectors.js#L608-L635